x-pack/plugins/security_solution/public/common/components/grouping/groups_selector/index.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  EuiContextMenuPanelDescriptor,
+  EuiContextMenuPanelItemDescriptor,
+} from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiBetaBadge, EuiPopover } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useMemo, useState } from 'react';
+import type { FieldSpec } from '@kbn/data-views-plugin/common';
+import { CustomFieldPanel } from './custom_field_panel';
+import { GROUP_BY, TECHNICAL_PREVIEW } from '../translations';
+import { StyledContextMenu, StyledEuiButtonEmpty } from '../styles';
+
+const none = i18n.translate('xpack.securitySolution.groupsSelector.noneGroupByOptionName', {
+  defaultMessage: 'None',
+});
+
+interface GroupSelectorProps {
+  fields: FieldSpec[];
+  groupSelected?: string;
+  onClearSelected: () => void;
+  onGroupChange: (groupSelection?: string) => void;
+  options: Array<{ key: string; label: string }>;
+  title?: string;
+}
+
+const GroupsSelectorComponent = ({
+  fields,
+  groupSelected = 'none',
+  onClearSelected,
+  onGroupChange,
+  options,
+  title = '',
+}: GroupSelectorProps) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const panels: EuiContextMenuPanelDescriptor[] = useMemo(
+    () => [
+      {
+        id: 'firstPanel',
+        items: [
+          {
+            'data-test-subj': 'panel-none',
+            name: none,
+            icon: groupSelected === 'none' || !groupSelected ? 'check' : 'empty',
+            onClick: onClearSelected,
+          },
+          ...options.map(
```

Comment:
```suggestion
          ...options.map<EuiContextMenuPanelItemDescriptor>(
```

Will get rid of the `as` cast on line `60`.